### PR TITLE
NR-35459-Flutter Web Quickstart.

### PR DESCRIPTION
**Pull request by pkudikyala**
# Summary

**Ticket URL:** https://issues.newrelic.com/browse/NR-35459

**Description:** This PR has the new quickstart integration named "Flutter Web"

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include an InstallPlan and Documentation reference?
- [ ] Are all documentation links publically accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [ ] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

## Screenshot

screencapture-newrelic-instant-observability-preview-2023-01-19-18_54_37

### CODE REVIEW

File: alert-policies/flutter-web/ClientandServerErrorResponse.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Description and details
+description: |+
+  This alert is triggered when the client and server error response is higher than 10 for 5 minutes.
+# Type of alert
+type: STATIC
+
+# NRQL query
+nrql:
+  query: "SELECT uniqueCount(requestUrl) FROM AjaxRequest WHERE httpResponseCode > 399 FACET requestUrl"
```
Comment (pkosanam) [inline alert-policies/flutter-web/ClientandServerErrorResponse.yml:12]: SELECT uniqueCount(requestUrl) FROM AjaxRequest WHERE httpResponseCode > 399

**Review by pkosanam — CHANGES_REQUESTED** [review]

**Comment by brammerl** [comment]
Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source.

If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

File: alert-policies/flutter-web/ApplicationErrorRate.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Name of the alert
+name: Application Error Rate
+
+# Description and details
+description: |+
+  This alert is triggered when the application error rate are higher than 5 for 5 minutes.
```
Comment (pkosanam) [inline alert-policies/flutter-web/ApplicationErrorRate.yml:6]: This alert is triggered when the application error rate is higher than 5 for 5 minutes.

**Review by pkosanam — CHANGES_REQUESTED** [review]

File: alert-policies/flutter-web/ClientandServerErrorResponse.yml (hunk 1 above)
Comment (pkudikyala) [reply alert-policies/flutter-web/ClientandServerErrorResponse.yml]: Hi @pkosanam,
Updated alert query. kindly review it.

File: alert-policies/flutter-web/ApplicationErrorRate.yml (hunk 2 above)
Comment (pkudikyala) [reply alert-policies/flutter-web/ApplicationErrorRate.yml]: Hi @pkosanam,
Updated alert description. kindly review it.

**Review by pkosanam — APPROVED** [review]
looks good

**Comment by zstix** [comment]
Hey @pkosanam, Thanks for the great PR! Like Lucia [mentioned above](https://github.com/newrelic/newrelic-quickstarts/pull/1594#issuecomment-1433583872), we are moving away from install plans in favor of data sources. Would you mind removing the install plan and making a data source instead?

File: quickstarts/flutter-web/config.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    description: |
+      Monitor the performance metrics of your Flutter Web application instances in real-time with New Relic Browser agent.
+
+    url: >-
+      https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/browser-monitoring-integrations/flutter-web-integration/
+keywords:
+  - flutter
+  - flutter web
+installPlans:
+  - flutter-web
```
Comment (zstix) [inline quickstarts/flutter-web/config.yml:43]: Let's add a reference to the newly created data source:

```suggestion
installPlans:
  - flutter-web
dataSourceIds
  - flutter-web
```

Comment (pkudikyala) [reply quickstarts/flutter-web/config.yml:43]: Hi @zstix ,
Added dataSourceId. Kindly review if need any changes let me know.

Thank you...

File: alert-policies/flutter-web/ApplicationErrorRate.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Name of the alert
+name: Application Error Rate
+
+# Description and details
+description: |+
+  This alert is triggered when the application error rate is higher than 5 for 5 minutes.
```
Comment (brammerl) [inline alert-policies/flutter-web/ApplicationErrorRate.yml:6]:
```suggestion
  This alert is triggered when the application error count is higher than 5 for 5 minutes.
```

File: data-sources/flutter-web/config.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+id: flutter-web
+displayName: Flutter Web
+description: |
```
Comment (brammerl) [inline data-sources/flutter-web/config.yml:3]: Data source descriptions don't support markdown and are expected to be a short description. Please come up with a short/concise description for this data source

File: data-sources/flutter-web/config.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  primary:
+    nerdlet:
+      nerdletId: nr1-install-newrelic.quickstart-installation-plan
+      nerdletState:
+        quickstartId: 
+      requiresAccount: true
+  fallback:
+    link:
+      url: https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/browser-monitoring-integrations/flutter-web-integration/
+keywords:
```
Comment (brammerl) [inline data-sources/flutter-web/config.yml:34]: If you want your quickstart to appear under a certain category in add data, please add specific keywords that will map to a specific category. The terms that appear match to the displayName.
[Query](https://api.newrelic.com/graphiql?#query=%7B%0A%20%20actor%20%7B%0A%20%20%20%20nr1Catalog%20%7B%0A%20%20%20%20%20%20categories%20%7B%0A%20%20%20%20%20%20%20%20terms%0A%20%20%20%20%20%20%20%20displayName%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A)

File: data-sources/flutter-web/config.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  Install and start monitoring your data today! 
+
+icon: logo.png
+install:
+  primary:
+    nerdlet:
+      nerdletId: nr1-install-newrelic.quickstart-installation-plan
+      nerdletState:
+        quickstartId:
```
Comment (brammerl) [inline data-sources/flutter-web/config.yml:29]: Seems like this is blank. Let's add a quickstartId!

File: data-sources/flutter-web/config.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+icon: logo.png
+install:
+  primary:
+    nerdlet:
+      nerdletId: nr1-install-newrelic.quickstart-installation-plan
+      nerdletState:
+        quickstartId: 
+      requiresAccount: true
+  fallback:
```
Comment (brammerl) [inline data-sources/flutter-web/config.yml:31]: I believe that this fallback should be the primary install. The current primary install points to the`quickstart-installation-plan` which isn't a proper destination for a datasource.

File: data-sources/flutter-web/config.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  
+  ## Comprehensive monitoring quickstart for Flutter Web.
+  
+  Monitor your Flutter applications, and analyze performance bottlenecks while operating in a live development environment. Visually monitor your web user experience across the world with your application rendition via the dashboard.
+
+  Install and start monitoring your data today! 
+
+icon: logo.png
+install:
+  primary:
```
Comment (brammerl) [inline data-sources/flutter-web/config.yml:25]:
```suggestion
  primary:
   link:
     url: https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/browser-monitoring-integrations/flutter-web-integration/
```

Comment (brammerl) [reply data-sources/flutter-web/config.yml:25]: Don't commit this but the primary install should be the link install instead I believe.

File: quickstarts/flutter-web/config.yml [hunk 10]
```diff
@@ -0,0 +1,49 @@
+
+
```
Comment (brammerl) [inline quickstarts/flutter-web/config.yml:2]: Let's remove this white space

File: alert-policies/flutter-web/ApplicationErrorRate.yml (hunk 4 above)
Comment (pkudikyala) [reply alert-policies/flutter-web/ApplicationErrorRate.yml]: Hi @brammerl ,

File: alert-policies/flutter-web/ClientandServerErrorResponse.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
+# Name of the alert
+name: Client and Server Error Response
+
+# Description and details
+description: |+
+  This alert is triggered when the client and server error response is higher than 10 for 5 minutes.
```
[the remark this answers is not in the document]
Comment (pkudikyala) [reply alert-policies/flutter-web/ClientandServerErrorResponse.yml]: Hi @brammerl ,
The changes are done. Could you review it and if need any changes let me know.
Thanks...

File: data-sources/flutter-web/config.yml (hunk 5 above)
Comment (pkudikyala) [reply data-sources/flutter-web/config.yml]: Hi @brammerl ,
I updated the description. Could you review it and if need any changes let me know.

Thanks...

File: data-sources/flutter-web/config.yml (hunk 8 above)
Comment (pkudikyala) [reply data-sources/flutter-web/config.yml]: Hi @brammerl ,
I removed the install plans file and updated the data source installplan. Could you review it and if need any changes let me know.

Thanks...

File: quickstarts/flutter-web/config.yml (hunk 10 above)
Comment (pkudikyala) [reply quickstarts/flutter-web/config.yml]: Hi @brammerl ,
Removed white spaces. Could you review it and if need any changes let me know.

Thanks...

File: quickstarts/flutter-web/config.yml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  - New Relic
+title: Flutter Web
+documentation:
+  - name: Flutter Web integration documentation
+    description: |
+      Monitor the performance metrics of your Flutter Web application instances in real-time with New Relic Browser agent.
+
+    url: >-
+      https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/browser-monitoring-integrations/flutter-web-integration/
+keywords:
```
[the remark this answers is not in the document]
Comment (pkudikyala) [reply quickstarts/flutter-web/config.yml:37]: Hi @brammerl ,
I updated the keywords. Could you review it and if need any changes let me know.

Thanks...

**Review by pkosanam — APPROVED** [review]
looks good

**Comment by pkudikyala** [comment]
The automated checks were not passed, we have created the below PR. Hence closing this PR
PR: https://github.com/newrelic/newrelic-quickstarts/pull/1716

**Pull request closed without merging**